tensorrt_llm/_torch/modules/fused_moe/impl_base.py

```
@@ -14,37 +14,145 @@
+
+
+def apply_moe_impl_construction_state(
+    module: nn.Module,
+    *,
+    routing_method: BaseMoeRoutingMethod,
+    num_experts: int,
+    hidden_size: int,
+    intermediate_size: int,
+    dtype: Optional[torch.dtype] = None,
+    reduce_results: bool = False,
+    model_config: ModelConfig = ModelConfig(),
```

Ruff (0.16.1)

[warning] Do not perform function call `ModelConfig` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)